pangeo_forge_recipes/transforms.py

```diff
@@ -634,7 +643,7 @@ def expand(
             | beam.combiners.Sample.FixedSizeGlobally(1)
             | beam.FlatMap(lambda x: x)  # https://stackoverflow.com/a/47146582
         )
-        # TODO: optionally use `singleton_target_store` to
-        # consolidate metadata and/or coordinate dims here
+        if self.consolidate_dimension_coordinates:
+            singleton_target_store = singleton_target_store | ConsolidateDimensionCoordinates()
```

Comment:
```suggestion
```

I went back and forth on this, but IMHO I think the right path here is to not do any consolidation by default, and leave it to users to compose modularly, i.e.:

```python

recipe = (
    ...
    | StoreToZarr()
    | ConsolidateCoordinateDimensions()
    | ConsolidateMetadata()
)
```

This is more verbose for users, but it is also more explicit as to what they can expect, and it adheres to the composable ethos of Beam, and it's less API-creep for us to maintain in StoreToZarr.

Happy to discuss dissenting views here, but as of now this feels like the right path to me.

Reply:
Would love to chat more about this! 
I also do like the composability, but wonder about the trade off for transformations that should be done on almost every recipe, but might get overlooked by a newer recipe developer.

Reply:
If you have a chat, I would love to attend if it fits on my schedule.

Reply:
Thanks for weighing in! I am not 100% sure which design is best here, and appreciate the lively discussion so we can land on the best solution. (Here and in the comment thread above.)

> done on almost every recipe, but might get overlooked by a newer recipe developer.

This is a small point, but one that just came to mind: in my experience, for smaller zarr stores, consolidation doesn't really matter / have a performance impact. Therefore, in some sense do we care if newer recipe developers (which presumably implies smaller-scale as well?) overlook this?

We absolutely should have a design discussion on this, possibly before the next coordination meeting (to expedite the process). I will try to first outline what I see as the various options in a comment on this thread to solicit async feedback.

Reply:
Awesome! Happy to chat before. I won't be able to make it to the next coordination meeting, so this week or next before Friday would be ideal for me.